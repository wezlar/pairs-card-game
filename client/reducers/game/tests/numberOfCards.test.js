import actionCreator from '../../../../testHelpers/actionCreator';

import ACTION_TYPES from '../../../actions/constants';
import reducer, { getInitialState } from '../numberOfCards';

describe('numberOfCards.js reducer', () => {
  const initialState = getInitialState();

  describe('initial state', () => {
    test('it matches the snapshot - default', () => {
      expect(initialState).toMatchSnapshot();
    });
  });

  describe('the reducer', () => {
    test('it returns initial state by default', () => {
      expect(reducer(undefined, {})).toEqual(initialState);
    });
    
    test('it doesn\'t change state unexpectedly', () => {
      const fakeAction = actionCreator('NOT_A_REAL_ACTION_TYPE')('test-payload');
      expect(reducer(undefined, fakeAction)).toEqual(initialState);
    });

    test(`status is correct for ${ACTION_TYPES.UPDATE_NUMBER_OF_CARDS}`, () => {
      const payload = { numberOfCards: 12 };
      const action = actionCreator(ACTION_TYPES.UPDATE_NUMBER_OF_CARDS)(payload);
      expect(reducer(undefined, action)).toEqual(12);
    });
  });
});