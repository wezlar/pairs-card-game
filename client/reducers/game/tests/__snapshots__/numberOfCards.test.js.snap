// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`numberOfCards.js reducer initial state it matches the snapshot - default 1`] = `16`;
